File: module_practice_ndk/src/main/java/com/alie/modulepracticendk/MainActivity.java
package com.alie.modulepracticendk;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

/**
 * Reference to learn ：
 * https://www.jianshu.com/p/4c890e23c021
 */

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivity";
    private NativeRawMgr mNativeRawMgr = NativeRawMgr.getInstance();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initData();
    }


    void initData() {
        mNativeRawMgr.doHelloWorld();
        mNativeRawMgr.doTest01();//【ndk print】int
        mNativeRawMgr.doTest02();//【ndk print】String
        mNativeRawMgr.doTest03();//【ndk print】int[]
        mNativeRawMgr.doTest04();//【ndk print】String[]
        mNativeRawMgr.doTest05();//【ndk print】Object - Cpu
        mNativeRawMgr.doTest06();//【ndk print】static method
        mNativeRawMgr.doTest07();//【ndk print】Object - Field
        mNativeRawMgr.doTest08();//【ndk create】Object - generate Cpu in ndk
    }
}

File: module_practice_common/src/main/java/com/alie/modulepracticecommon/MainActivity.kt
package com.alie.modulepracticecommon

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import com.alie.modulepracticecommon.util.AvoidDoubleClickListener

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        startMethod()
    }

    /**
     * fast click test
     */
    private fun doTest01() {
        println("===doTest01")
        findViewById<Button>(R.id.btn1).setOnClickListener(object : AvoidDoubleClickListener() {
            override fun onViewClick(v: View?) {
                println("===onViewClick go go go")
            }
        })
    }

    private fun startMethod() {
        doTest01()//【fast click】 fast click test
    }


}